mpi4jax/_src/collective_ops/allgather.py

```diff
@@ -169,7 +169,7 @@ def mpi_allgather_abstract_eval(x, token, comm):
     out_shape = (size, *x.shape)
     return (
         abstract_arrays.ShapedArray(out_shape, x.dtype),
```

Comment:
`ShapedArray` is also defined in `jax.core`. Should we stop importing from `abstract_arrays` entirely?

Reply:
Who knows. 
Ill try to open an issue on jax to know what they are going to do

Reply:
Yes, good idea. We should probably merge in the meantime?

Reply:
Yes I’ll merge and tag 0.3.1